collectors/ebpf.plugin/ebpf.c

```diff
@@ -681,36 +681,48 @@ static void parse_port_list(void **out, char *range)
 {
     ebpf_network_viewer_port_list_t **list = (ebpf_network_viewer_port_list_t **)out;
 
+    char *copied = strdupz(range);
     char *end = range;
     //Move while I cannot find a separator
     while (*end && *end != ':' && *end != '-') end++;
 
     //It has a range
+    int first, last;
     if (likely(*end)) {
         *end++ = '\0';
+        last = str2i((const char *)end);
+    } else {
+        last = 0;
     }
 
-    int test;
-    test = str2i((const char *)range);
-    if (test < NETDATA_MINIMUM_PORT_VALUE || test > NETDATA_MAXIMUM_PORT_VALUE) {
-        info("The port %d is invalid and it will be ignored!", test);
+    first = str2i((const char *)range);
+    if (first < NETDATA_MINIMUM_PORT_VALUE || first > NETDATA_MAXIMUM_PORT_VALUE) {
+        info("The first port %d of the range \"%s\" is invalid and it will be ignored!", first, copied);
+        freez(copied);
         return;
     }
 
-    ebpf_network_viewer_port_list_t *w = callocz(1, sizeof(ebpf_network_viewer_port_list_t));
-    w->value = strdup(range);
-    w->hash = simple_hash(range);
-
-    w->first = (uint16_t)htons((uint16_t)test);
-    test = ((likely(*end)))?str2i((const char *)end):test;
-    if (test < NETDATA_MINIMUM_PORT_VALUE || test > NETDATA_MAXIMUM_PORT_VALUE) {
-        info("The second port %d of the range is invalid and the whole range will be ignored!", test);
-        freez(w->value);
-        freez(w);
+    if (!last)
+        last = first;
+
+    if (last < NETDATA_MINIMUM_PORT_VALUE || last > NETDATA_MAXIMUM_PORT_VALUE) {
+        info("The second port %d of the range \"%s\" is invalid and the whole range will be ignored!", last, copied);
+        freez(copied);
         return;
     }
 
-    w->last = (uint16_t)htons((uint16_t)test);
+    if (first > last) {
+        info("The specified order [%d, %d] is inverted, Netdata is fixing it!", first, last);
```

Comment:
i dont think this is correct

example:

`333-100`

It is likely an error is `100` => should be `333-1000`. 

What u are doing here is creating a range based on your assimptions.

A user didnt ask to create `100-333` range.

Reply:
Let user to fix it.

Ranges should reflect configuration file. At the moment they are not.

Reply:
my point

`333-100`

The error likely:
- additional digit in the first, fixed: `33-100`
- missing digit in the last, fixed: `333-1000`


It is unlikely the user wants `100-333`

Reply:
Yeah I think ilyam is right every case of `-INT` `INT-` and reversed order is most likely misplaced space... maybe we should error in every case instead of trying to be "`smart`"

Reply:
I got your point of view, I pushed a commit that does not add the value to the list if the order is wrong.